analyzers/src/SonarAnalyzer.CSharp/Rules/InsecureHashAlgorithm.cs

```diff
@@ -82,8 +84,10 @@ public sealed class InsecureHashAlgorithm : DoNotCallInsecureSecurityAlgorithmBa
                 KnownType.System_Security_Cryptography_HMACRIPEMD160
             );
 
-        protected override Location Location(ObjectCreationExpressionSyntax objectCreation) =>
-            objectCreation.Type.GetLocation();
+        protected override Location Location(SyntaxNode objectCreation) =>
+            objectCreation is ObjectCreationExpressionSyntax objectCreationExpression
```

Comment:
Why not to reuse the factory here as well? This `if` logic (including error handling) is already implemented there.

Reply:
We agreed to do it in slightly different way, since there is no Type part in `ImplicitObjectCreationException`